pages/calldata-decoder.page.tsx

```diff
@@ -0,0 +1,91 @@
+import { Fragment, ParamType } from '@ethersproject/abi'
+import { BigNumber } from '@ethersproject/bignumber'
+import { zip } from 'lodash'
+import { ChangeEvent, useState } from 'react'
+
+import { decodeCalldata } from '../lib/decodeCalldata'
+import { parseAbi } from '../lib/parseAbi'
+
+type Decoded = string | BigNumber
+
+export default function CalldataDecoder() {
```

Comment:
Nice, it works! Few points: 

<img width="1023" alt="image" src="https://user-images.githubusercontent.com/1814312/158352394-920237a9-c71f-4302-a5ca-6701aee80283.png">

1. Please make the whole design wider. 
2. Delete sample data encoded in the view
3. Add error handling is missing, but maybe we can do it later as part of the bigger story regarding form validation.

Reply:
4. function sig (4 bytes) is missing as we discussed.